src/components/program-progress/ProgramProgressCourses.jsx

```diff
@@ -0,0 +1,149 @@
+import React, { useContext } from 'react';
```

Comment:
Will this component capture all of the checks mentioned in this utility https://github.com/openedx/edx-platform/blob/master/lms/static/js/learner_dashboard/models/course_card_model.js#L142-L181

Reply:
yes, it is